### ISSUE TITLE
Not pulling Citaiton Counts at all (Semantic Scholar) or for books (CrossRef)

### ISSUE THREAD

**Issue by LReinoso1** [eschnett/zotero-citationcounts#20]
I installed the add-on to track citation counts for articles in order to prioritize.  I noticed that the add-on doesn't pull citation counts even for works that have such citation counts (e.g. [Michael Saward's _The Representative Claim_](https://www.semanticscholar.org/search?q=saward%20the%20representative%20claim&sort=relevance)).  Likewise, the add-on seems not to collect book citations for CrossRef.  The add-on pulls CrossRef citations for the article "The Representative Claim", but not the book by the same title -- despite [CrossRef recording citations for the book](https://api.crossref.org/v1/works/10.1093/acprof:oso/9780199579389.001.0001).

**Comment by eschnett** [comment]
Is there an error message?
Does it say that it found 0 citations, or is there any other pop-up message?
Does it work when you manually trigger updating the citation counts by right-clicking on the publication?

**Comment by LReinoso1** [comment]
In both cases, it just says that it found 0 citations.  More specifically "Semantic Scholar/CrossRef citation counts updated for 0 items."

Manual triggering doesn't help.

**Comment by LReinoso1** [comment]
I am wondering if there is any follow-up.  Thank you!

**Comment by eschnett** [comment]
I have no idea what is causing this problem. I will have to debug what is going wrong – I can't promise a particular timeline.

**Comment by LReinoso1** [comment]
Thank you!

**Comment by craigl-res** [comment]
I'm getting the same issue with each citation source. Happy to beta test a fix if you find one. 
Edit: I'm using Zotero 6.0.5 and Windows 11

**Comment by craigl-res** [comment]
An update: the plugin appears to work fine (and well) for Zotero records with a DOI entry, and returns the '0 item' message for records without a DOI entry. I haven't tested every record I have but that appears to be a pattern so far.

**Comment by LReinoso1** [comment]
Thanks!  Is that for both Semantic Scholar and CrossRef?  I am pulling articles on CrossRef, but the plugin pulls no Semantic Scholar citations

**Comment by LReinoso1** [comment]
Also, I just checked: if I have a DOI for a book, it still does not pull CrossRef citation counts

**Comment by craigl-res** [comment]
I tested on one article and it can pull the CrossRef citations too, if a valid DOI is present.

**Comment by RakhDamir** [comment]
I have encountered the same problem. For some references from ArXiv, I do not get any information about the number of citations. It does not show even 0 citations. When I try to trigger the update manually it shows that " ... citation counts were updated for 0 items".

**Comment by LReinoso1** [comment]
CrossRef works for articles, not for books (even if I have DOI in "extra" field).  Semantic Scholar doesn't work for anything.

**Comment by glocalglocal** [comment]
> the add-on seems not to collect book citations for CrossRef

CrossRef only deals with journals, not books. AFAIK, this addon doesn't support counts for books at all unfortunately.

**Comment by Leonpagliacci** [comment]
the same problem arose with the pop-up message "Semantic Scholar/CrossRef citation counts updated for 0 items." when I first installed this plug-in.

**Comment by adrien-perello** [comment]
I agree, I think it is most likely is because books and book chapters do not have a "DOI" field.

In `In zoterocitationcounts.js` there is:

```js
async function getSemanticScholarCount(item, idtype) {
    let doi = null;
    if (idtype == 'doi') {
        doi = item.getField('DOI');
    } else if (idtype == 'arxiv') {
        const arxiv = item.getField('url'); // check URL for arXiv id
        const patt = /(?:arxiv.org[/]abs[/]|arXiv:)([a-z.-]+[/]\d+|\d+[.]\d+)/i;
        const m = patt.exec(arxiv);
        if (!m) {
            // No arxiv id found
            return -1;
        }
        doi = m[1];
    } else {
        // Internal error
        return -1;
    }
    if (!doi) {
        // There is no DOI / arXiv id; skip item
        return -1;
    }
```

With a bit of regex, we can get the DOI from either the "URL" field or the "extra" (as it is often added there). Which is what @eschnett does, but It seems to me that he assumes that if there is no DOI, then the only reason could be because it is not printed yet, and then he checks for arxiv. But in my case for example the DOI reference is present in the "URL" field (as well as the "extra" field since it is added automatically with the "Save to Zotero" browser plugin).
